src/query/functions/aggregation/quantile.go

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package aggregation
+
+import (
+	"math"
+	"sort"
+)
+
+const (
+	// QuantileType takes the n-th non nan quantile element in a list of series
+	// Special cases are:
+	// 	 n < 0 = -Inf
+	// 	 n > 1 = +Inf
+	QuantileType = "quantile"
+)
+
+// Creates a quantile aggregation function for a given q-quantile measurement
+func makeQuantileFn(opType string, q float64) (aggregationFn, bool) {
+	if opType != QuantileType {
+		return nil, false
+	}
+	return func(values []float64, bucket []int) float64 {
+		return quantileFn(q, values, bucket)
+	}, true
+}
+
+func quantileFn(q float64, values []float64, bucket []int) float64 {
```

Comment:
can we split this out since part where we get bucketVals and then later part which creates actual quantile ? That way we can move the second thing out and reuse it at other place (or use a library later)